Remove deprecated streams and schema changes for existing streams

rdeshmukh15 opened the pull request:
# Description of change
- Schama changes as per the API doc [link](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#overview)
- Removed schemas for 2 streams **_meeting_files_** and  _**webinar_files**_  because the respective api endpoints are deprecated ([Ref link](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#tag/Deprecated-API-Endpoints:~:text=archived%20file%20statistics-,Deprecated%20API%20Endpoints,-Groups))
- field name changes :
         - **_meeting_registrants_** : field name `county` changed to `country`
         - **_meetings_** : 
                     1. field name `encrypted_Password` changed to `encrypted_password`
                     2. field name `occurences` changed to `occurrences`
                     3. field name `occurence_id` changed to `occurrence_id`
          - **_webinar_absentees_** : field name `county` changed to `country`
          - **_webinar_registrants_** : field name `county` changed to `country`
         - **_webinars_** : 
                     1. field name `occurences` changed to `occurrences`
                     2. field name `occurence_id` changed to `occurrence_id`
- field datatype changes :
         -**_meetings_** :  
                     1. _pmi_ datatype changed from `integer` to `string`
                     2. _contact_name_ datatype changed from `boolean` to `string`
                     3. _contact_email_ datatype changed from `boolean` to `string`
                     4. _weekly_days_ datatype changed from `integer` to `string`
          -**_report_meeting_participants_** :  _duration_ datatype changed from `string` to `integer`
           -**_report_webinar_participants_** :  _duration_ datatype changed from `string` to `integer`
          -**_webinars_** : 
                     1. _id_ datatype changed from `string` to `integer`
                     2. _contact_name_ datatype changed from `boolean` to `string`
                     3. _contact_email_ datatype changed from `boolean` to `string`
                     4. _weekly_days_ datatype changed from `integer` to `string`
- removed deprecated fields :
       -**_meetings_** :  enforce_login, enforce_login_domains
       -**_webinars_** :  enforce_login, enforce_login_domains
- Added new fields in schemas meeting_poll_results, meeting_polls, meeting_registrants, report_meetings, report_webinars, users, webinar_panelists, webinar_poll_results, webinar_polls

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

rdeshmukh15 committed:
schema changes

kethan1122 commented on tap_zoom/schemas/meetings.json:
```diff
@@ -193,6 +253,12 @@
+            "boolean"
+          ]
+        },
         "cn_meeting": {
```
This field(`cn_meeting`) seems to be deprecated and also `in_meeting`.

[reference](https://devforum.zoom.us/t/cn-meeting-and-in-meeting/4420)

kethan1122 commented on tap_zoom/schemas/report_meetings.json:
```diff
@@ -104,6 +104,32 @@
+            ]
+          }
+        }
+      }
```
I see few fields like `has_screen_share `, `host_organization `, `has_recording` there are many actually which are missing. Did we miss them or Am I looking at the wrong [doc](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/reportMeetings)

rdeshmukh15 replied:
Yes, looking at the wrong doc. The endpoint is report/meetings/{meeting_id} and the doc is https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/reportMeetingDetails

kethan1122 commented on tap_zoom/schemas/report_webinar_participants.json:
```diff
@@ -46,14 +46,38 @@
+        "integer"
       ]
     },
     "attentiveness_score": {
```
is this a valid field(`attentiveness_score`). I don't see it in [docs](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/reportWebinarParticipants)

kethan1122 commented on tap_zoom/schemas/report_webinars.json:
```diff
@@ -116,6 +116,32 @@
         "null",
         "string"
       ]
+    },
```
`user_id` field is not mentioned in [docs](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/reportWebinarDetails). are we generating during tap run ?

rdeshmukh15 replied:
Yes, we are generating it during tap run

kethan1122 commented on tap_zoom/schemas/users.json:
```diff
@@ -102,6 +114,56 @@
         "integer"
       ]
+    },
+    "user_created_at": {
```
on line 70 we have `created_at` which seems to be a deprecated field [doc](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/users)

kethan1122 commented on tap_zoom/schemas/webinar_poll_results.json:
```diff
@@ -40,6 +40,19 @@
+            "type": [
+              "null",
+              "string"
+            ]
```
are `start_time` and `webinar_id` fields not required for this stream

kethan1122 replied:
https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/listPastWebinarPollResults

rdeshmukh15 replied:
No, as you can see in endpoints.py file, for webinar_poll_results path data_key is `questions` and those are only the part of api response we are storing.

kethan1122 commented on tap_zoom/schemas/webinars.json:
```diff
@@ -212,6 +403,12 @@
+            "null",
+            "boolean"
+          ]
+        },
```
`close_registration` seems to be a deprecated field [doc](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/webinar)

rdeshmukh15 committed:
review comments addressed

rdeshmukh15 committed:
changes :
- added new line at the end of the file
- added null type as well for object type of items

shantanu73 committed:
[TDL-22406] Add integration tests (#16)

* schema changes

* review comments addressed

* added empty bradcrumb to the catalog

* added forced-replication-method to the empty breadcrumb

* changes in endpoints_config

* Preparing integration test suite.

* Added env vars to fetch to run tests

* Changes:
1) Removed start date from config properties as all the streams are full table.
2) Added RESPECTS_START_DATE parameter in expected metadata.
3) Added discovery test.

* Added test all fields

* Modified discovery test.

* Modified test all fields.

* Added integration tests for all fields, automatic fields & pagination

* Added page limit for test pagination

* added empty bradcrumb to the catalog

* added forced-replication-method to the empty breadcrumb

* changes in endpoints_config

* Removed deprecated streams from integration tests.

* Added config to run integration tests

* Fixed assertions & mmodifications after test runs

* Modified CircleCI config to run integration tests parellely.

* Added executor for integration tests.

* Fixed CircleCI config

* Addedd relevant streams and assertions for all fields test

* Changes:
1) Modified CirclCI config to run integration tests.
2) Fixed assertions & streams to test for test_pagination.

* Minor fixtures

* Changed streams to test for Automatic Test

* Minor fixes and comments.

* Fixed review comments

* Resolved review comments. Removed redundant code

* Removed unused method

* Fixed comments in the test files.

* Added comment to point to defect to skip wev=binars stream in pagination test.

* Added descriptive comments for excluding webinars stream.

* Dictionary implementation for missing fields per stream.

Co-authored-by: Rushikesh Todkar <98420315+RushiT0122@users.noreply.github.com>

* Fixed keys in MISSING_FIELDS dictionary.

* Fixed logic for MISSING_FIELDS for all fields test

* schema changes

* review comments addressed

* added empty bradcrumb to the catalog

* added forced-replication-method to the empty breadcrumb

* changes in endpoints_config

* Preparing integration test suite.

* Added env vars to fetch to run tests

* Changes:
1) Removed start date from config properties as all the streams are full table.
2) Added RESPECTS_START_DATE parameter in expected metadata.
3) Added discovery test.

* Added test all fields

* Modified discovery test.

* Modified test all fields.

* Added integration tests for all fields, automatic fields & pagination

* Added page limit for test pagination

* added empty bradcrumb to the catalog

* added forced-replication-method to the empty breadcrumb

* changes in endpoints_config

* Removed deprecated streams from integration tests.

* Added config to run integration tests

* Fixed assertions & mmodifications after test runs

* Modified CircleCI config to run integration tests parellely.

* Added executor for integration tests.

* Fixed CircleCI config

* Addedd relevant streams and assertions for all fields test

* Changes:
1) Modified CirclCI config to run integration tests.
2) Fixed assertions & streams to test for test_pagination.

* Minor fixtures

* Changed streams to test for Automatic Test

* Minor fixes and comments.

* Fixed review comments

* Resolved review comments. Removed redundant code

* Removed unused method

* Fixed comments in the test files.

* Added comment to point to defect to skip wev=binars stream in pagination test.

* Added descriptive comments for excluding webinars stream.

* Dictionary implementation for missing fields per stream.

Co-authored-by: Rushikesh Todkar <98420315+RushiT0122@users.noreply.github.com>

* Fixed keys in MISSING_FIELDS dictionary.

* Fixed logic for MISSING_FIELDS for all fields test

* fix integration tests

* Fixed zoom tap-tester integration tests.

---------

Co-authored-by: “rdeshmukh15” <rdeshmukh@talend.com>
Co-authored-by: Rushikesh Todkar <98420315+RushiT0122@users.noreply.github.com>
Co-authored-by: RushiT0122 <rtodkar@stitchdata-talend.com>

shantanu73 committed:
Changed expected page size to 2
